Cleanup integration tests post-Zuul migration

jillr opened the pull request:
##### SUMMARY
This collection has been operating on Zuul CI for some weeks now.
- Remove shippable references
- Use shorter unique_ids were resource_prefixes now exceed AWS resource length limits (zuul hostnames are longer than shippable was)
- Cleanup for CI bugs surfaced in the course of this PR

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
shippable.yml
tests/
README.md

NOT-Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/978

jillr commented:
There's a number of tests that build a shortened `unique_id` from the `resource_prefix` based on what these were with shippable.  eg; tests/integration/targets/sns_topic/defaults/main.yml  
Zuul does not append a test-run unique digit at the end of the prefix like shippable did.  We're not currently doing any parallel tests, so we could get by with a single set of digits from these tests, but curious what you think @tremble before I make changes.

tremble commented:
@jillr someone's already started just md5sum-ing the resource_prefix I'm good with that approach.

tremble reviewed (approved):
LGTM

tremble commented:
recheck

jillr commented:
recheck

jillr commented:
the only changes since the last review are from cleaning up a merge conflict, and then fixing a line I messed up in that conflict resolution, will gate once CI passes on the existing review so it doesn't go into conflict again.

jillr commented:
recheck

jillr commented:
recheck

jillr commented:
Working my way through resource names that are length exceeded with the zuul resource_prefix length; since we don't have ci_complete jobs anymore this is effectively the first time we've run into most of these.  It's overloading this PR a bit but my priority right now is making CI green while we have a change to test that actually touches every target.

goneri commented:
Since this PR will likely trigger the whole test-suite, I would suggest to split it up in 3 or 4 PR. This way it will be easier to merge the smaller PR one by one.

jillr commented:
recheck

goneri commented:
recheck

jillr commented:
recheck

jillr commented:
recheck

tremble commented on tests/integration/targets/aws_codepipeline/defaults/main.yml:
```diff
@@ -1,12 +1,9 @@
 # IAM role names have to be less than 64 characters
-# The 8 digit identifier at the end of resource_prefix helps determine during
-# which test something was created and allows tests to be run in parallel
-# Shippable resource_prefixes are in the format shippable-123456-123, so in those cases
-# we need both sets of digits to keep the resource name unique
-unique_id: "{{ resource_prefix | regex_search('(\\d+-?)(\\d+)$') }}"
+# we hash the resource_prefix to get a shorter, unique string
+unique_id: "{{ resource_prefix | hash('md5') | truncate(6, True, '') }}"
```
Niggle - use unique_id in here?

jillr replied:
Good catch - thanks

tremble commented on tests/integration/targets/lambda_alias/defaults/main.yml:
```diff
@@ -1,10 +1,7 @@
-unique_id: "{{ resource_prefix | regex_search('(\\d+-?)(\\d+)$') }}"
+# we hash the resource_prefix to get a shorter, unique string
+unique_id: "{{ resource_prefix | hash('md5') }}"
+lambda_function_name: '{{ unique_id }}'
```
Slight concern here: this is going to result in a load of lambda resources with just an MD5 has as the ID, and nothing to tie them back to ansible tests.  While this is fine for Ansible's CI, this may cause confusion in private test environments.

tremble commented on tests/integration/targets/redshift/defaults/main.yml:
```diff
@@ -1,6 +1,7 @@
 # defaults file for test_redshift
-redshift_cluster_name: '{{ resource_prefix }}'
+unique_id: "{{ resource_prefix | hash('md5') }}"
+redshift_cluster_name: '{{ unique_id }}'
```
```suggestion
redshift_cluster_name: 'ansible-test-{{ unique_id }}'
```

md5sums can start with a number.  For Database names this is not allowed: they must start with a letter

tremble commented on tests/integration/targets/elb_target/playbooks/roles/elb_lambda_target/defaults/main.yml:
```diff
@@ -1,5 +1,5 @@
-lambda_name: '{{ resource_prefix }}-elb-target'
-elb_target_group_name: '{{ resource_shortprefix }}-elb-tg'
+lambda_name: '{{ unique_id }}-elb-target'
+elb_target_group_name: '{{ unique_id | truncate(8, True, '') }}-elb-tg'
```
```suggestion
elb_target_group_name: "{{ unique_id | truncate(8, True, '') }}-elb-tg"
```

jillr commented:
recheck

tremble commented:
recheck

tremble commented:
recheck

tremble commented:
recheck

tremble commented:
recheck

jillr commented:
recheck

goneri commented:
recheck

tremble commented:
recheck

tremble commented:
recheck

tremble commented:
recheck

tremble commented:
And finally, all tests passed!  The latest changes to the way things are distributed have worked.

tremble commented:
Previous test worked, rebased to remove the merge commit

tremble commented:
@goneri with https://github.com/ansible/ansible-zuul-jobs/pull/962 and https://github.com/ansible/ansible-zuul-jobs/pull/979 we finally have passing tests, any chance we can get everything merged?

jillr commented:
gate hit some of the usual API lag issues, retrying

goneri commented:
I see the gating is failing again. I try with my "splitter" branch.

goneri commented:
recheck

goneri commented:
recheck

goneri commented:
recheck

goneri commented:
recheck

goneri commented:
recheck

goneri commented:
recheck

goneri commented:
recheck

tremble commented:
recheck

tremble commented:
recheck
